File: List in Python/Linked_List.py
class Node:
    def __init__(self, x = 0, n = None):
        self.val = x
        self.next = n

class Linked_List:
    def __init__(self):
        self.sz = 0
        self.head = None
    def insert(self, x):
        if(self.sz == 0):
            self.head = Node()
            self.tail = self.head
            self.head.val = x
            self.tail.next = None
        else:
            new_node = Node()
            self.tail.next = new_node
            self.tail = self.tail.next
            self.tail.val = x
        self.sz += 1
        
    
    def erase_end(self):
        if(self.sz == 0):
            return None
        
        if(self.head == self.tail):
            self.tail = None
        else:
            cur = self.head
            while cur.next is not self.tail:
                cur = cur.next
            self.tail = cur
            self.tail.next = None
        self.sz -= 1
        
        
    def erase_front(self):
        if(self.sz == 0):
            return None
        if(self.head == self.tail):
            self.head = None
            self.tail = None
        else:
            cur = self.head
            self.head = self.head.next
            cur = None
        self.sz -= 1
            
    def size(self):
        return self.sz
    def display(self):
        cur = self.head
        while cur != None:
            print(cur.val, end=' ')
            cur = cur.next


l = Linked_List()
print("Size: ", l.size())
l.insert(5)
print("Size: ", l.size())
l.insert(4)
print("Size: ", l.size())
l.insert(6)
print("Size: ", l.size())
l.display()
print()
l.erase_end()
print("Size: ", l.size())
l.display()
l.erase_front()
print("Size: ", l.size())
l.display()

File: List In Java/src/Linked_List.java
class Node{
    public int info;
    public Node next;
    public Node prev;
    public Node(){
        next = null;
        prev = null;
        info = 0;
    }
}

public class Linked_List {
    private Node head;
    private Node tail;
    private int _size;
    public Linked_List(){
        head = null;
        tail = null;
        _size = 0;
    }
    void insert(int x){
        if(_size == 0){
            head = new Node();
            head.info = x;
            tail = head;
        }else{
            tail.next = new Node();
            tail.next.info = x;
            tail.next.prev = tail;
            tail = tail.next;
        }
        _size++;
    }
    void erase(){
        if(head == tail){
            head = null;
            tail = null;
            _size--;
        }
        else if(head == null){
            System.err.println("List is Empty");
        }
        else{
            tail = tail.prev;
            tail.next.prev = null;
            tail.next = null;
            _size--;
        }
    }
    int front(){
        if(head == null){
            System.err.println("List is Empty");
            //return 0;
        }
        return head.info;
    }
    int back(){
        if(head == null){
            System.err.println("List is Empty");
            return 0;
        }
        return tail.info;
    }
}
